Resolves #3583 - Session fast forward enabled when COM_BINLOG_DUMP_GTID is received

jaredev opened the pull request:
Resolves [Issue 3583](https://github.com/sysown/proxysql/issues/3583)

Enables session fast forwarding if the COM_BINLOG_DUMP_GTID command is received.

jaredev committed:
_MYSQL_COM_BINLOG_DUMP_GTID command

* Added COM_BINLOG_DUMP_GTID command to enum_mysql_commands with value 30 (0x1e) based on https://dev.mysql.com/doc/internals/en/com-binlog-dump-gtid.html

jaredev committed:
Receiving COM_BINLOG_DUMP_GTID enabled session fast forwaring

* If the COM_BINLOG_DUMP_GTID is received, a warning is logged, and then the session_fast_foward is set to true

jaredev commented:
Here are the basic changes for this. TBD if any cleanup or other changes are needed when changing the state of the session in this way. Feedback appreciated.

renecannao commented:
This needs a way for testing it consistently.
Probably it is enough to create a test that starts mysqlbinlog with `--read-from-remote-server`, using the replication user and connecting to proxysql where the same user is **not** configured with `fast_forward`

pondix commented:
Automated message: PR pending admin approval for build testing

javsanpar committed:
Commands that read the binlog enable fast forward

For commands BINLOG_DUMP, BINLOG_DUMP_GTID and REGISTER_SLAVE a lot of
the features provided by ProxySQL aren't useful, like multiplexing,
query parsing, etc. For this reason, ProxySQL enable fast_forward when
it receives these commands.

javsanpar committed:
Add test checking COM_BINLOG_DUMP enables fast_forward

Commit:
fix calling mysqlbinlog for the CI

mirostauder commented:
retest this please

javsanpar committed:
Add test checking COM_REGISTER_SLAVE enables fast_forward

This test runs a custom binlog client, that uses libslave, to test that
ProxySQL activates fast forward automatically after receiving the
command, and to test that it can read the binlog events. The custom
client reads the binlog after sending a query that do not disables
multiplexing and after sending a query that disables multiplexing.

mirostauder commented:
retest this please

JavierJF committed:
Fix 'connect_timeout' and 'connect_retries' for 'MYSQL_COM_BINLOG_DUMP'

Fix honoring of 'connect_timeout_server_max' and
'connect_retries_on_failure' for 'fast_forward' sessions created by
'MYSQL_COM_BINLOG_DUMP' when session still doesn't own a backend
connection.

JavierJF committed:
Improve comments for 'MYSQL_COM_BINLOG_DUMP' session preparation
